src/NUnitTestAdapterTests/NuspecDependenciesTests.cs

```diff
@@ -0,0 +1,312 @@
+﻿// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NUnit.Framework;
+
+#nullable enable
+
+namespace NUnit.VisualStudio.TestAdapter.Tests
+{
+    /// <summary>
+    /// Checks that the csproj of the framework projects are reflected correctly in the nuspec files.
+    /// </summary>
+    [TestFixture("NUnit3TestAdapter.nuspec", "NUnitTestAdapter/NUnit.TestAdapter.csproj")]
+    internal sealed class NuspecDependenciesTests
+    {
+        private readonly string _nuspecPath;
+        private readonly string _csprojPath;
+        private readonly string _propsPath;
+
+        private Dictionary<string, List<PackageWithVersion>> _nuspecPackages;
+        private Dictionary<string, List<PackageWithVersion>> _csprojPackages;
+        private Dictionary<string, string> _csprojPackageVersions;
+
+        private const string Root = "../../../../../";
+        private const string NotSpecified = nameof(NotSpecified);
+
```

Comment:
Suggest adding a list with the packages that are development usage only, and those we should ignore due to they being embedded.